hazelcast/src/test/java/com/hazelcast/ringbuffer/RingbufferMergeIntegerValuesMergePolicy.java

```diff
@@ -0,0 +1,46 @@
+package com.hazelcast.ringbuffer;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.spi.impl.merge.RingbufferMergeData;
+import com.hazelcast.spi.merge.SplitBrainMergePolicy;
+import com.hazelcast.spi.merge.SplitBrainMergeTypes.RingbufferMergeTypes;
+
+class RingbufferMergeIntegerValuesMergePolicy
+        implements SplitBrainMergePolicy<RingbufferMergeData<Object>, RingbufferMergeTypes> {
+
+    @Override
+    public RingbufferMergeData<Object> merge(RingbufferMergeTypes mergingValue, RingbufferMergeTypes existingValue) {
```

Comment:
Can we make it fail-fast if not Integer data type?

Reply:
Actually this merge policy is used to select out integer values from a mixed type (Integer & String) ringbuffers, both on the existing and the merging side. We could make it throw an exception, of course. How would the test look then?